.github/ISSUE_TEMPLATE/bug_report.md

```diff
@@ -10,23 +10,70 @@ assignees: ''
 
 The release number or commit hash of the version you're using.
 
-## Operating system & version
+## \[Optional\] Setup information
 
-Ex: Windows 10, Ubuntu 18.04, iOS 14.
+If you are reporting a bug about:
+
+- difficulties getting Bevy to build or run on your machine
+- unusual rendering bugs
+- unusual input bugs
+- hardware specific problems
+
+These issues are often specific to hardware or OS configuration and can be challenging to reproduce.
+As a result, please consider including information about:
+
+- the Rust version you're using
+  - Bevy has a policy of relying on the "latest stable release" of Rust
+  - Nightly should generally work, but there are sometimes regressions: please let us know!
+- the operating system, including its version
+  - e.g. Windows 10, Ubuntu 18.04, iOS 14.
+- relevant hardware
+  - e.g. Nvidia GTX 1080 graphics card or XBox series X controller
+
+You should also consider testing the examples of our upstream dependencies to help isolate the issue:
+
+- [`wgpu`](https://github.com/gfx-rs/wgpu) for rendering problems
+- [`winit`](https://github.com/rust-windowing/winit) for input and window management
+- [`gilrs`](https://docs.rs/gilrs/latest/gilrs/) for gamepad inputs
 
 ## What you did
 
-The steps you took to uncover this bug. Please list full reproduction steps if
-feasible.
+The steps you took to uncover this bug.
+Please provide a runnable snippet that demonstrates the problem if feasible.
 
-## What you expected to happen
+For example:
```

Comment:
I'm not sure it's necessary to show an actual code snippet. I prefer keeping those type of templates as short as possible.

Reply:
I can definitely see the value in that.

However, this is targeted at absolute beginners: this may be their first open source interaction, and teaching them how to make "minimal reproducible examples" is worth the time IMO.

Reply:
No, hard disagree. A bug report is a bug report. Its job is to get relevant info from users so we can track and fix an issue, not to teach users how to collaborate on open-source projects.